File: python/LeetCode/03637.py
from typing import List

def isTrionic(self, nums: List[int]) -> bool:
    n = len(nums)

    if nums[1] - nums[0] < 0:
        return False

    curr_direction = False
    count = 0
    for i in range(n - 1):
        v1 = nums[i]
        v2 = nums[i + 1]

        if v2 - v1 == 0:
            return False

        direction = v2 - v1 > 0
        if direction != curr_direction:
            count += 1
            curr_direction = direction

        if count > 3:
            return False

    return count == 3

File: go/leetcode/03637.go
package main

func isTrionic(nums []int) bool {
	n := len(nums)
	i := 1

	for i < n && nums[i-1] < nums[i] {
		i += 1
	}
	p := i - 1

	for i < n && nums[i-1] > nums[i] {
		i += 1
	}
	q := i - 1

	for i < n && nums[i-1] < nums[i] {
		i += 1
	}
	flag := i - 1

	return (p != 0) && (q != p) && (flag == n-1 && flag != q)
}
